### PR TITLE
fix: app.js export can be non-function

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: test/fixtures/boot/app/plugin/boot-plugin-empty/app.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+'use strict';
+
+module.exports = 'not function';
```
Comment (fengmk2) [inline test/fixtures/boot/app/plugin/boot-plugin-empty/app.js]: 额，按道理原来也会报错。

File: lib/loader/mixin/custom.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ module.exports = {
       if (is.class(bootHook)) {
         // if is boot class, add to lifecycle
         this.lifecycle.addBootHook(bootHook);
-      } else {
+      } else if (is.function(bootHook)) {
```
Comment (fengmk2) [inline lib/loader/mixin/custom.js]: 加个 warning

File: test/fixtures/boot/app/plugin/boot-plugin-empty/app.js (hunk 1 above)
Comment (dead-horse) [reply test/fixtures/boot/app/plugin/boot-plugin-empty/app.js]: https://github.com/eggjs/egg-core/blob/4.10.0/lib/loader/egg_loader.js#L306 之前的兼容在这里

**Comment by fengmk2** [comment]
4.10.2

**Pull request merged**